crates/sui-graphql-rpc/src/types/transaction_block/tx_lookups.rs

```diff
@@ -0,0 +1,468 @@
+            ),
+        });
+    }
+    if let Some(kind) = &filter.kind {
+        subqueries.push((select_kind(*kind, tx_bounds), "tx_kinds"));
+    }
+    if let Some(recv) = &filter.recv_address {
+        subqueries.push((select_recipient(recv, sender, tx_bounds), "tx_recipients"));
+    }
+    if let Some(input) = &filter.input_object {
+        subqueries.push((select_input(input, sender, tx_bounds), "tx_input_objects"));
+    }
+    if let Some(changed) = &filter.changed_object {
+        subqueries.push((
+            select_changed(changed, sender, tx_bounds),
+            "tx_changed_objects",
+        ));
+    }
+    if let Some(sender) = &sender {
+        if filter.complex_filters() == 0 || filter.kind.is_some() {
+            subqueries.push((select_sender(sender, tx_bounds), "tx_senders"));
+        }
+    }
+    if let Some(txs) = &filter.transaction_ids {
+        subqueries.push((select_ids(txs, tx_bounds), "tx_digests"));
+    }
+
+    if subqueries.is_empty() {
+        return None;
+    }
+
+    let mut subquery = subqueries.pop().unwrap().0;
+
+    if !subqueries.is_empty() {
+        subquery = query!("SELECT tx_sequence_number FROM ({}) AS initial", subquery);
+        while let Some((subselect, alias)) = subqueries.pop() {
+            subquery = inner_join!(subquery, alias => subselect, using: ["tx_sequence_number"]);
+        }
+    }
+
+    Some(subquery)
+}
+
+pub(crate) fn select_tx(sender: Option<SuiAddress>, bound: TxBounds, from: &str) -> RawQuery {
+    let mut query = query!(format!("SELECT tx_sequence_number FROM {}", from));
+
+    if let Some(sender) = sender {
+        query = filter!(query, format!("sender = {}", bytea_literal(&sender)));
+    }
+
+    filter!(
+        query,
+        format!(
+            "tx_sequence_number >= {} AND tx_sequence_number <= {}",
+            bound.lo, bound.hi
+        )
+    )
+}
+
+pub(crate) fn select_pkg(
+    pkg: &SuiAddress,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_calls_pkg"),
+        format!("package = {}", bytea_literal(pkg))
+    )
+}
+
+pub(crate) fn select_mod(
+    pkg: &SuiAddress,
+    mod_: String,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_calls_mod"),
+        format!("package = {} and module = {{}}", bytea_literal(pkg)),
+        mod_
+    )
+}
+
+pub(crate) fn select_fun(
+    pkg: &SuiAddress,
+    mod_: String,
+    fun: String,
+    sender: Option<SuiAddress>,
+    bound: TxBounds,
+) -> RawQuery {
+    filter!(
+        select_tx(sender, bound, "tx_calls_fun"),
+        format!(
+            "package = {} AND module = {{}} AND func = {{}}",
+            bytea_literal(pkg),
+        ),
+        mod_,
+        fun
+    )
+}
+
+pub(crate) fn select_kind(kind: TransactionBlockKindInput, bound: TxBounds) -> RawQuery {
+    filter!(
+        select_tx(None, bound, "tx_kinds"),
```

Comment:
Why `None` here? If the kind is programmable, then it's valid to additionally filter it with a `sender` -- it's just that it becomes a `select_sender` query if the sender is not `0x0` (or an empty query otherwise).

Reply:
Right, in tandem with your comment below, we can leverage the sender handling here (and thus simplify the logic further up the stack)